test/test_type.jl

```diff
@@ -1852,6 +1852,320 @@ isdir(outdir) && rm(outdir, recursive = true)
+                                                                        orientation,
+                                                                        direction,
+                                                                        x, t,
+                                                                        surface_flux_function,
+                                                                        equations)) == RealT
+                end
+
+                @test eltype(@inferred Trixi.calc_wavespeed_roe(u_ll, u_rr, direction,
+                                                                equations)) ==
+                      RealT
+            end
+
+            @test eltype(@inferred flux(u, orientation, equations)) == RealT
+            @test eltype(@inferred flux_nonconservative_wintermeyer_etal(u_ll, u_rr,
+                                                                         orientation,
+                                                                         equations)) ==
+                  RealT
+            @test eltype(@inferred flux_nonconservative_fjordholm_etal(u_ll, u_rr,
+                                                                       orientation,
+                                                                       equations)) == RealT
+            @test eltype(@inferred flux_nonconservative_audusse_etal(u_ll, u_rr,
+                                                                     orientation,
+                                                                     equations)) == RealT
+            @test eltype(@inferred flux_nonconservative_ersing_etal(u_ll, u_rr, orientation,
+                                                                    equations)) == RealT
+            @test eltype(@inferred flux_fjordholm_etal(u_ll, u_rr, orientation,
+                                                       equations)) == RealT
+            @test eltype(@inferred flux_wintermeyer_etal(u_ll, u_rr, orientation,
+                                                         equations)) == RealT
+
+            @test eltype(eltype(@inferred hydrostatic_reconstruction_audusse_etal(u_ll,
+                                                                                  u_rr,
+                                                                                  equations))) ==
+                  RealT
+
+            @test typeof(@inferred max_abs_speed_naive(u_ll, u_rr, orientation, equations)) ==
+                  RealT
+            @test eltype(@inferred dissipation(u_ll, u_rr, orientation, equations)) == RealT
+            @test eltype(@inferred dissipation(u_ll, u_rr, normal_direction, equations)) ==
+                  RealT
+            @test eltype(@inferred numflux(u_ll, u_rr, orientation, equations)) == RealT
+            # no matching method 
+            # @test eltype(@inferred numflux(u_ll, u_rr, normal_direction, equations)) == RealT
```

Comment:
Try to test both `orientation` and `normal_direction` but it showed
``` julia
MethodError: no method matching min_max_speed_davis(::StaticArraysCore.SVector{3, Float32}, ::StaticArraysCore.SVector{3, Float32}, ::StaticArraysCore.SVector{1, Float32}, ::Trixi.ShallowWaterEquations1D{Float32})
```

Reply:
Is that okay to comment like this? @ranocha

Reply:
Yes, that's fine 👍